xbmc/network/NetworkServices.cpp

```diff
@@ -88,8 +88,9 @@ using namespace UPNP;
 using KODI::MESSAGING::HELPERS::DialogResponse;
 
 CNetworkServices::CNetworkServices()
+  : m_settings(CServiceBroker::GetSettingsComponent()->GetSettings()),
```

Comment:
if not required by a compiler warning
imo for better readability was better where was before

Reply:
iIt's about performance. Instead of first initializing automatically m_settings by calling the default ctor and after this assigning another value right in the ctor body, we could just init m_settings with the value here, so saving one assignment, thus being more performant. That's the point here for the cppcheck suggestion/warning.

Reply:
now got it better thanks, sometimes this cppcheck suggestion/warning is boring